Podcast/SearchITunesTableViewController.swift

```diff
@@ -0,0 +1,123 @@
+        request.success = { request in
+            guard let results = request.processedResponseValue as? [Series] else { return }
+            if results.isEmpty {
+                self.tableView.backgroundView?.isHidden = false
+            }
+            self.searchResults = results
+            self.tableView.reloadData()
+            self.loadingIndicatorView?.stopAnimating()
+        }
+        request.failure = { request in
+            self.tableView.backgroundView?.isHidden = false
+            self.loadingIndicatorView?.stopAnimating()
+        }
+        System.endpointRequestQueue.addOperation(request)
+        completion?()
+    }
+    
+    // MARK: TableView
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return seriesCellHeight
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if let cell = tableView.dequeueReusableCell(withIdentifier: seriesCellIdentifier) as? SearchSeriesTableViewCell {
+            cell.configure(for: searchResults[indexPath.row], index: indexPath.row)
+            cell.delegate = self
+            return cell
+        }
+        let seriesCell = SearchSeriesTableViewCell(style: .default, reuseIdentifier: seriesCellIdentifier)
+        seriesCell.configure(for: searchResults[indexPath.row], index: indexPath.row)
+        seriesCell.delegate = self
+        return seriesCell
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchResults.count
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let seriesDetailViewController = SeriesDetailViewController(series: searchResults[indexPath.row])
+        navigationController?.pushViewController(seriesDetailViewController,animated: true)
+    }
+    
+    // MARK: UISearchBarDelegate
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        guard let searchText = searchBar.text else { return }
+        fetchData(query: searchText, completion: nil)
+    }
+    
+    func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        tableView.isHidden = false
+        tableView.backgroundView?.isHidden = true
+    }
+    
+    // MARK: UISearchResultsUpdating
+    
+    func updateSearchResults(for searchController: UISearchController) {
+        tableView.backgroundView?.isHidden = true
+    }
+    
+    // MARK: SearchSeriesCellDelegate
+    
+    func searchSeriesTableViewCellDidPressSubscribeButton(cell: SearchSeriesTableViewCell) {
+        guard let indexPath = tableView.indexPath(for: cell) else { return }
+        let series = searchResults[indexPath.row]
+        series.subscriptionChange(completion: cell.setSubscribeButtonToState(isSubscribed:numberOfSubscribers:))
```

Comment:
same here with the `series.subscriptionChange(completion: cell.setSubscribeButtonToState)`